code/WorkInProgress/JobXPRewards.dm

```diff
@@ -499,3 +499,25 @@ mob/verb/checkrewards()
 		banana.set_loc(get_turf(C.mob))
 		C.mob.put_in_hand(banana)
 		return
+
+/////////////Barman////////////////
+
+/datum/jobXpReward/barman/goldenshaker
+	name = "Golden Cocktail Shaker"
+	desc = "After all your years of service, you've finally managed to gather enough money in tips to buy yourself a present! You regret every cent."
+	required_levels = list("Barman"=20)
+	icon_state = "?"
+	claimable = 1
+	claimPerRound = 1
+	var/path_to_spawn = /obj/item/reagent_containers/food/drinks/cocktailshaker/golden
+
+	activate(var/client/C)
+		var/obj/item/reagent_containers/food/drinks/cocktailshaker/shaker = locate(/obj/item/reagent_containers/food/drinks/cocktailshaker) in C.mob.contents
+
+		if (istype(shaker))
+			C.mob.remove_item(shaker)
+			qdel(shaker)
+		var/obj/item/I = new path_to_spawn()
+		I.set_loc(get_turf(C.mob))
+		C.mob.put_in_hand_or_drop(I)
```

Comment:
This looks like it would materialize a shaker out of thin air if they weren't holding one, no?

Reply:
Should be resolved now